src/utils/tasks.js

```diff
@@ -0,0 +1,42 @@
+import TaskState from '@/model/TaskState.model'
+
+/**
+ * States used when the parent is stopped.
+ * @type {*[]}
+ */
+const isStoppedOrderedStates = [
+  TaskState.SUBMIT_FAILED,
+  TaskState.FAILED,
+  TaskState.RUNNING,
+  TaskState.SUBMITTED,
+  TaskState.EXPIRED,
+  TaskState.READY,
+  TaskState.SUBMIT_RETRYING,
+  TaskState.RETRYING,
+  TaskState.SUCCEEDED,
+  TaskState.QUEUED,
+  TaskState.WAITING,
+  TaskState.HELD,
+  TaskState.RUNAHEAD
+]
+
+/**
+ * Gives a single state, based on a list of states of children nodes.
+ * @param childStates {Array} children nodes
+ * @param isStopped {Boolean} whether the parent node is stopped or not
+ * @returns {null|*} a valid Task State, or null if not found
+ * @link @see https://github.com/cylc/cylc-flow/blob/d66ae5c3ce8c749c8178d1cd53cb8c81d1560346/lib/cylc/task_state_prop.py
+ */
```

Comment:
I guess this fixes us to a given version of cylc-flow?

Reply:
Yup, I think @hjoliver sent me the link on Riot to the the file in a branch for 7.x. Then I used it to write the JS code, and pressed "y" on my keyboard while focusing on the GitHub UI. That creates this link to the specific commit.